generator/internal/discovery_file.cc

```diff
@@ -65,37 +65,29 @@ Status DiscoveryFile::FormatFile(std::string const& product_name,
 syntax = "proto3";
 
 package $package_name$;
-
 )""");
 
   if (!import_paths_.empty()) {
-    printer.Print(vars, absl::StrJoin(import_paths_, "\n",
-                                      [](std::string* s, std::string const& p) {
-                                        *s += absl::StrFormat("import \"%s\";",
-                                                              p);
-                                      })
-                            .c_str());
-    printer.Print("\n\n");
+    printer.Print("\n");
+    for (auto const& path : import_paths_) {
+      printer.Print(vars, absl::StrFormat("import \"%s\";\n", path).c_str());
```

Comment:
Observation: It appears that newer protobuf releases (v22.x) have changed the `Print()` argument from `const char*` to `absl::string_view`, so we'll be able to get rid of all the `.c_str()` cruft soon-ish.

Reply:
Neat.